File: 8kyu/multiple-of-index/multiple-of-index.py
# def multiple_of_index(arr):
#     res = []
#     for i, num in enumerate(arr):
#         if i != 0 and num % i == 0:
#             res.append(num)
#     return res

def multiple_of_index(arr):
    return [num for idx, num in enumerate(arr) if idx and num % idx == 0]

print(multiple_of_index([22, -6, 32, 82, 9, 25]))

File: 8kyu/multiple-of-index/multiple-of-index.go
package main

import "fmt"

func multipleOfIndex(ints []int) []int {
	// var res []int
	// res := make([]int, 0)
	res := []int{}

	for index, num := range ints {
		if index != 0 && num%index == 0 {
			res = append(res, num)
		}
	}

	return res
}

func main() {
	fmt.Println(multipleOfIndex([]int{22, -6, 32, 82, 9, 25}))
}
